src/components/searchBar.jsx

```diff
@@ -0,0 +1,48 @@
+import { TextField, InputAdornment, IconButton } from '@mui/material';
+import { useState } from "react"
+import { useSearchParams } from "react-router-dom";
+import SearchIcon from '@mui/icons-material/Search';
+
+export default function SearchBar() {
+    let [searchParams, setSearchParams] = useSearchParams();
+    const [searchTerm, setSearchTerm] = useState('');
+
+    const handleInputChange = (e) => {
+        setSearchTerm(e.target.value);
+    };
+
+    function handleSubmit(event) {
+        event.preventDefault();
```

Comment:
The `event` given by the form will have the text field value so there is no need to create a state for that. Let's also redirect to a new page, something like `/search?q=something` instead of adding the URL params to this page